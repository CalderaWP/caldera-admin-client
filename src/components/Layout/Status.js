import React from 'react';
import {Spinner} from '@wordpress/components';
import PropTypes from 'prop-types'
import {Admin} from '@caldera-labs/components'

export const Status = (props) => {
	if (props.updating) {
		return (
			<Spinner/>
		)
	}
	return (
		<Admin.StatusIndicator
			message={props.message}
			show={props.show}
			success={props.success}
		/>
	)
};

Status.propTypes = {
	updating: PropTypes.bool,
	show: PropTypes.bool,
	success: PropTypes.bool,
	message: PropTypes.string,
};

Status.defaultProps = {
	show: false,
	success: true,
};


