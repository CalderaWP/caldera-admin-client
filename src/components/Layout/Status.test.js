import React from 'react';
import renderer from "react-test-renderer";
import {Status} from "./Status";


describe( 'Status component', () => {
	it( 'Shows spinner when updating', () => {
		const component = renderer.create(<Status
			updating={true}
		/>);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'Shows message when it should', () => {
		const component = renderer.create(<Status
			updating={false}
			message={'Hi Roy'}
			show={true}
		/>);
		expect(component.toJSON()).toMatchSnapshot();
	});
});