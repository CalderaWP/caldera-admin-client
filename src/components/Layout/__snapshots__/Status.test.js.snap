// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component Shows message when it should 1`] = `
<div
  className="cf-alert-wrap"
>
  <p
    className="cf-alert cf-alert-success"
  >
    Hi Roy
  </p>
</div>
`;

exports[`Status component Shows spinner when updating 1`] = `
<span
  className="spinner is-active"
/>
`;
